daemon/api_users.go

```diff
@@ -477,6 +484,18 @@ func getUserDetailsFromAssertion(st *state.State, modelAs *asserts.Model, email
 	if len(su.Models()) > 0 && !strutil.ListContains(su.Models(), model) {
 		return "", nil, fmt.Errorf(errorPrefix+"%q not in models %q", model, su.Models())
 	}
+	// XXX: should we really be this paranoid here, format check
+	// is already done on the assertion level
+	if len(su.Serials()) > 0 && su.Format() > 0 {
```

Comment:
no, the su.Format() bit is actually confusing here. Places outside of asserts shouldn't really care about that